src/Ionide.ProjInfo/Library.fs

```diff
@@ -764,10 +767,30 @@ type WorkspaceLoaderViaProjectGraph private (toolsPath, ?globalProperties: (stri
             pg
 
     let loadProjects (projects: ProjectGraph, customProperties: string list, binaryLogs: BinaryLogGeneration) =
+        let handleError (e: exn) =
+            let msg = e.Message
+            printfn "error -> %A" e
+            logger.error (Log.setMessage "Failed loading" >> Log.addExn e)
+
+            projects.ProjectNodesTopologicallySorted
+            |> Seq.distinctBy (fun p -> p.ProjectInstance.FullPath)
+            |> Seq.iter (fun p ->
+
+                let p = p.ProjectInstance.FullPath
+
+                if msg.Contains "The project file could not be loaded." then
+                    loadingNotification.Trigger(WorkspaceProjectState.Failed(p, ProjectNotFound(p)))
+                elif msg.Contains "not restored" then
+                    loadingNotification.Trigger(WorkspaceProjectState.Failed(p, ProjectNotRestored(p)))
+                else
+                    loadingNotification.Trigger(WorkspaceProjectState.Failed(p, GenericError(p, msg))))
+
+            Seq.empty
+
         try
             lock WorkspaceLoaderViaProjectGraph.locker
             <| fun () ->
-                let allKnown = projects.ProjectNodesTopologicallySorted |> Seq.distinctBy (fun p -> p.ProjectInstance.FullPath)
+                let allKnown = projects.ProjectNodes |> Seq.distinctBy (fun p -> p.ProjectInstance.FullPath)
```

Comment:
I've seen `ProjectNodesTopologicallySorted` throw in the past for unknown reasons.